src/daemon/rpmostreed-os.cxx

```diff
@@ -1639,32 +1629,49 @@ os_handle_get_cached_deploy_rpm_diff (RPMOSTreeOS *interface,
                                                   version,
                                                   cancellable,
                                                   &checksum,
-                                                  &local_error))
-        goto out;
+                                                  error))
+        return glnx_prefix_error (error, "Looking up cached version");
     }
 
   if (!rpm_ostree_db_diff_variant (ot_repo, base_checksum, checksum, FALSE, &value,
-                                   cancellable, &local_error))
-    goto out;
+                                   cancellable, error))
+    return glnx_prefix_error (error, "Assembling diff");
 
   details = rpmostreed_commit_generate_cached_details_variant (base_deployment,
                                                                ot_repo,
                                                                rpmostree_origin_get_refspec (origin),
                                                                checksum,
-                                                               &local_error);
-  if (!details)
-    goto out;
+                                                               error);
+  if (details == NULL)
+    return glnx_prefix_error (error, "Generating cached details");
 
-out:
+  *out_value = util::move_nullify (value);
+  *out_details = util::move_nullify (details);
+  return TRUE;
+}
+
+static gboolean
+os_handle_get_cached_deploy_rpm_diff (RPMOSTreeOS *interface,
+                                      GDBusMethodInvocation *invocation,
+                                      const char *arg_revision,
+                                      const char * const *arg_packages)
+{
+  GError *local_error = NULL;
+  g_autoptr(GVariant) value = NULL;
+  g_autoptr(GVariant) details = NULL;
+
+  /* XXX Ignoring arg_packages for now. */
+  gboolean is_ok = get_cached_deploy_rpm_diff (interface, arg_revision,
+                                               &value, &details, &local_error);
   if (local_error != NULL)
-    {
-      g_dbus_method_invocation_take_error (invocation, local_error);
-    }
-  else
-    {
-      g_dbus_method_invocation_return_value (invocation,
-                                             new_variant_diff_result (value, details));
-    }
+    g_dbus_method_invocation_take_error (invocation, local_error);
```

Comment:
I think this hunk should be part of the below `if (!is_ok)` conditional and instead we have `g_assert (local_error)` or so.

The conditionals are checking for the same thing - an error return.

Reply:
Ack, done.